specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,283 @@
+# Background
+
+We are exposing an event that will be raised when an attempt to launch an external protocol is made. The host will be given the option to cancel the launch, hide the dialog, as well as disable the checkbox that, if selected, will give permissions to always allow the launch.
+
+# Description
+
+This event will be raised before the external protocol launch occurs. Currently a dialog is displayed in which the user can click `Open` or `Cancel`. If the request is made from a [trustworthy origin](https://w3c.github.io/webappsec-secure-contexts/#potentially-trustworthy-origin) a checkmark box will be displayed that will allow the user to always allow this external protocol from this origin. The `NavigationStarting`, `NavigationCompleted`, `SourceChanged`, `ContentLoading`, and `HistoryChanged` events will not be raised when a request is made to launch an external protocol.
+
+The `LaunchingExternalProtocol` event will be raised on either `CoreWebView2` or `CoreWebView2Frame` depending on if the launch request is originating from the main frame or a non-main frame. In the case of a nested iframe requesting the external protocol launch, the event will be raised from the top level iframe.
```

Comment:
Yes, adding support for nested iframes is already planned.
From other side we already have such weird API add_PermissionRequested, which covers nested iframes on top-level iframe =(

Reply:
CoreWebView2.PermissionRequested covers top level document and all frames, correct? This sentence sounds like CoreWebView2.LaunchingExternalProtocol covers top level document, not child frames, but all other frames which is very confusing. If for this scenario we don't care so much about which frame has raised the event then we can move this to just CoreWebView2.LaunchingExternalProtocol and it is raised for top level document and all frames. If we do care about the frame context, then we can wait for nested iframes work to be done and have the event on the corresponding CoreWebView2/Frame.

Reply:
For the record what we have:
1. CoreWebView2Frame.PermissionRequested - event is raised for top-level iframe and all nested iframes;
2. CoreWebView2.PermissionRequested - event is raised for main page and all iframes including nested if they were not handled by 1)

Reply:
Sorry I don't think I made this clear earlier, the LaunchingExternalProtocol event follows the same structure as the permission requested frame event.  CoreWebView2Frame.LaunchingExternalProtocol fires for top level iframes/nested frames and CoreWebView2.LaunchingExternalProtocol is raised for the top level document and nested iframes if not handled by the frame event. The original ask was only to differentiate between if the request was originating from the main frame or iframe. Monarch originally asked for this, but since hasn't followed up about any feedback.

Reply:
Another option is because there is no formal ask for this anymore from Monarch, we can fire the event only from CoreWebView2 for now and later add support for iframes when nested iframes are supported/ if there is a request for iframes.